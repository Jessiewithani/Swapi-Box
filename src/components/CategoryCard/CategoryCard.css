.card {
    background-color: white;
    border: 2px solid black;
    display: flex;
    flex-direction: column;
    font-family: 'Dunkin';
}

.starBackground {
    background-image: url("../../images/Star.png");
    border: red 2px solid;
}

footer {
    display: inline-block;
}

header, 
ul, 
footer {
    justify-content: space-between;
}