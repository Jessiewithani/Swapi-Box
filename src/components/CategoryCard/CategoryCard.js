import React from 'react';
import '../CategoryCard/CategoryCard.css'

const CategoryCard = () => {
    return <h1>hello category card</h1>
}

export default CategoryCard;