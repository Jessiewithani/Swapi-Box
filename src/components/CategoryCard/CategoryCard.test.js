import React from 'react';
import { shallow } from 'enzyme';
import CategoryCard from './CategoryCard';

describe('CategoryCard', () => {
  const card = {name: 'Luke Skywalker', homeworld: 'Tatooine', species: 'Human', population: '200000'}
  const index = 1
  it('should match the snapshot', () => {
    const wrapper= shallow(<CategoryCard 
      key={index} 
      name={card.name} 
      homeworld= {card.homeworld} 
      species={card.species} 
      population={card.population} 
      model={card.model} 
      vehicle_class={card.vehicle_class} 
      passengers={card.passengers} 
      terrain={card.terrain} 
      climate={card.climate}
      residents={card.resident}/>)

      expect(wrapper).toMatchSnapshot()
  });
  it('should match the snapshot', ()=> {
    const card= {model: 'Digger Crawler', class: 'wheeled', passengers: '30'}
    const index= 1;
    const wrapper= shallow(<CategoryCard 
      key={index} 
      name={card.name} 
      homeworld= {card.homeworld} 
      species={card.species} 
      population={card.population} 
      model={card.model} 
      vehicle_class={card.vehicle_class} 
      passengers={card.passengers} 
      terrain={card.terrain} 
      climate={card.climate}
      residents={card.resident}/>)

      expect(wrapper).toMatchSnapshot()
  })
  it('should match the snapshot', ()=> {
    const card= {name: 'Hoth', terrain: 'tundra, ice cave, mountain ranges', climate: 'frozen', residents: []}
    const index= 1;
    const wrapper= shallow(<CategoryCard 
      key={index} 
      name={card.name} 
      homeworld= {card.homeworld} 
      species={card.species} 
      population={card.population} 
      model={card.model} 
      vehicle_class={card.vehicle_class} 
      passengers={card.passengers} 
      terrain={card.terrain} 
      climate={card.climate}
      residents={card.resident}/>)

      expect(wrapper).toMatchSnapshot()
  })
});