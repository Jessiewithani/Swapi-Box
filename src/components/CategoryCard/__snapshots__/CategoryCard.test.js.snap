// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard should match the snapshot 1`] = `
<div
  className="card"
>
  <header>
    <h3
      className="cardTitle"
    >
      <img
        alt="star"
        className="star"
        height="21"
        src="Star.png"
      />
      Luke Skywalker
    </h3>
  </header>
  <div
    className="list-container"
  >
    <ul
      className="cardList"
    >
      <li>
        Homeworld: 
        Tatooine
      </li>
      <li>
        Species: 
        Human
      </li>
      <li>
        Population: 
        200000
      </li>
    </ul>
  </div>
  <footer>
    <button
      className="cardButton"
      onClick={[Function]}
    >
      <img
        alt="heart"
        className="faveHeart"
        src="Heart.png"
      />
      FAV
    </button>
  </footer>
</div>
`;

exports[`CategoryCard should match the snapshot 2`] = `
<div
  className="card"
>
  <header>
    <h3
      className="cardTitle"
    >
      <img
        alt="star"
        className="star"
        height="21"
        src="Star.png"
      />
    </h3>
  </header>
  <div
    className="list-container"
  >
    <ul
      className="cardList"
    >
      <li>
        Model: 
        Digger Crawler
      </li>
      <li>
        Passengers: 
        30
      </li>
    </ul>
  </div>
  <footer>
    <button
      className="cardButton"
      onClick={[Function]}
    >
      <img
        alt="heart"
        className="faveHeart"
        src="Heart.png"
      />
      FAV
    </button>
  </footer>
</div>
`;

exports[`CategoryCard should match the snapshot 3`] = `
<div
  className="card"
>
  <header>
    <h3
      className="cardTitle"
    >
      <img
        alt="star"
        className="star"
        height="21"
        src="Star.png"
      />
      Hoth
    </h3>
  </header>
  <div
    className="list-container"
  >
    <ul
      className="cardList"
    >
      <li>
        Terrain: 
        tundra, ice cave, mountain ranges
      </li>
      <li>
        Climate: 
        frozen
      </li>
    </ul>
  </div>
  <footer>
    <button
      className="cardButton"
      onClick={[Function]}
    >
      <img
        alt="heart"
        className="faveHeart"
        src="Heart.png"
      />
      FAV
    </button>
  </footer>
</div>
`;
